internal/core/config/sources/file/file.go

```diff
@@ -21,6 +21,13 @@ type Config struct {
 	PollRateSeconds int `yaml:"pollRateSeconds" default:"5"`
 }
 
+// New creates a new file remote config source from the target config
+func (c *Config) New() (types.ConfigSource, error) {
+	return New(time.Duration(c.PollRateSeconds) * time.Second), nil
+}
```

Comment:
It is done because there is no generic way to otherwise instantiate the config sources without listing them all out and having a bunch of duplicated code.  It allows https://github.com/signalfx/signalfx-agent/pull/542/files#diff-80d94174b8b7f9702d3f815d80ad347bL57 and all of the common validation logic to be shared.